vendor/bat-native-ledger/src/bat/ledger/internal/promotion.cc

```diff
@@ -381,6 +384,11 @@ void Promotion::OnGetAllPromotions(
 
   if (list.size() > 0) {
     for (auto & item : list) {
+      if (item->legacy_claimed  ) {
```

Comment:
I think we need to do if else here and then at the end do insert as we don't want to add legacy promotions to UI. if we do this then we can change status to `ATTESTED`

Reply:
Done (by @NejcZdovc)